### PR TITLE
docs: updated examples to new api

### PR DESCRIPTION
**Pull request by achingbrain**
BREAKING CHANGE:

Where `PeerID`s were previously [CID]s, now they are Strings

- `ipfs.bitswap.stat().peers[n]` is now a String (was a CID)
- `ipfs.dht.findPeer().id` is now a String (was a CID)
- `ipfs.dht.findProvs()[n].id` is now a String (was a CID)
- `ipfs.dht.provide()[n].id` is now a String (was a CID)
- `ipfs.dht.put()[n].id` is now a String (was a CID)
- `ipfs.dht.query()[n].id` is now a String (was a CID)
- `ipfs.id().id` is now a String (was a CID)
- `ipfs.id().addresses[n]` are now [Multiaddr]s (were Strings)

Depends on:

- [x] https://github.com/ipfs/interface-js-ipfs-core/pull/581
- [x] https://github.com/ipfs/js-ipfs-http-client/pull/1226
- [x] https://github.com/libp2p/js-libp2p/pull/545

[CID]: https://www.npmjs.com/package/cids
[Multiaddr]: https://www.npmjs.com/package/multiaddr

### CODE REVIEW

**Comment by alanshaw** [comment]
Please add BREAKING CHANGE message.

File: examples/browser-video-streaming/index.html [hunk 1]
```diff
@@ -2,7 +2,11 @@
   <body>
     <video id="video" controls></video>
     <script src="../../dist/index.js"></script>
+    <!--
+      TODO: uncomment once https://github.com/ipfs/js-ipfs/pull/2683 and
+      https://github.com/moshisushi/hlsjs-ipfs-loader/pull/16 are released
     <script src="https://unpkg.com/hlsjs-ipfs-loader@0.1.4/dist/index.js"></script>
+    -->
```
Comment (alanshaw) [inline examples/browser-video-streaming/index.html]: Can now be uncommented?

File: examples/browser-video-streaming/streaming.js [hunk 2]
```diff
@@ -20,3 +19,145 @@ document.addEventListener('DOMContentLoaded', async () => {
     hls.on(Hls.Events.MANIFEST_PARSED, () => video.play())
   }
 })
+
+// TODO: remove everything below here once https://github.com/ipfs/js-ipfs/pull/2683
+// and https://github.com/moshisushi/hlsjs-ipfs-loader/pull/16 are released
```
Comment (alanshaw) [inline examples/browser-video-streaming/streaming.js]: Remove now?

File: examples/exchange-files-in-browser/public/app.js [hunk 3]
```diff
@@ -44,20 +44,30 @@ let info
...
   if (!node) {
-    const options = {
+    node = await IPFS.create({
       repo: 'ipfs-' + Math.random(),
       config: {
         Addresses: {
-          Swarm: ['/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star']
+          Swarm: [
+            // This is a public webrtc-star server
+            // '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star'
```
Comment (alanshaw) [inline examples/exchange-files-in-browser/public/app.js]:
```suggestion
            // '/dns4/star-signal.cloud.ipfs.team/wss/p2p-webrtc-star'
```

File: test/core/interface.spec.js [hunk 4]
```diff
@@ -23,7 +23,16 @@ describe('interface-ipfs-core tests', function () {
       ref: require('ipfs-http-client')
     },
     ipfsOptions: {
-      pass: 'ipfs-is-awesome-software'
+      pass: 'ipfs-is-awesome-software',
+      ...(isBrowser ? {
+        config: {
+          Addresses: {
+            Swarm: [
+              '/ip4/127.0.0.1/tcp/14579/wss/p2p-webrtc-star'
```
Comment (alanshaw) [inline test/core/interface.spec.js]:
```suggestion
              '/ip4/127.0.0.1/tcp/14579/ws/p2p-webrtc-star'
```

File: test/utils/factory.js [hunk 5]
```diff
@@ -18,6 +19,19 @@ const factory = (options, overrides) => createFactory(
...
     js: {
       ipfsBin: './src/cli/bin.js'
+    },
+    proc: {
+      ...(isBrowser ? {
+        ipfsOptions: {
+          config: {
+            Addresses: {
+              Swarm: [
+                '/ip4/127.0.0.1/tcp/14579/wss/p2p-webrtc-star'
```
Comment (alanshaw) [inline test/utils/factory.js]:
```suggestion
                '/ip4/127.0.0.1/tcp/14579/ws/p2p-webrtc-star'
```

File: examples/browser-video-streaming/streaming.js (hunk 2 above)
Comment (achingbrain) [reply examples/browser-video-streaming/streaming.js]: The `hlsjs-ipfs-loader` PR has been merged but not published yet so I don't think we can remove this yet.

File: examples/browser-video-streaming/index.html (hunk 1 above)
Comment (achingbrain) [reply examples/browser-video-streaming/index.html]: See the other comment, the `hlsjs-ipfs-loader` PR hasn't been published yet

File: examples/browser-video-streaming/streaming.js (hunk 2 above)
Comment (alanshaw) [reply examples/browser-video-streaming/streaming.js]: It got published! `0.2.3`

**Pull request merged**